apps/wallet/src/ui/app/redux/slices/transaction-requests/index.ts

```diff
@@ -65,10 +66,105 @@ export const loadTransactionResponseMetadata = createAsyncThunk<
     }
 );
 
+export type TxnMetaResponse = {
+    coinSymbol?: string | null;
+    amount?: number | null;
+    objectId?: string;
+    isListing?: boolean;
+} | null;
+
+const getRequestCost = (
+    txEffects: TransactionEffects,
+    address: string
+): TxnMetaResponse | null => {
+    const events = txEffects?.events || [];
+
+    let sumCoin = 0;
+    const coinMeta: {
+        objectId?: string;
+        amount?: number;
+        coinSymbol?: string;
+        isListing?: boolean;
+    } = {};
+
+    for (const event of events) {
+        if (
+            'coinBalanceChange' in event ||
+            'transferObject' in event ||
+            'moveEvent' in event
+        ) {
+            // Get the amount in move events
+            if (
+                event.moveEvent?.sender === address &&
+                event.moveEvent?.fields?.price
+            ) {
+                sumCoin += event.moveEvent.fields.price;
+                coinMeta.amount = sumCoin;
+                coinMeta.coinSymbol = '0x2::sui::SUI';
+                coinMeta.isListing = true;
+            }
+
+            // aggregate coin balance for pay
+            if (
+                event.coinBalanceChange?.changeType === 'Pay' &&
+                event.coinBalanceChange?.owner.AddressOwner === address
+            ) {
+                sumCoin += event.coinBalanceChange?.amount || 0;
```

Comment:
Do we know that if we have coinBalanceChange event of type Pay that we wont have a moveEvent?

If we can have then in one tx then we can mix different type of coins here.

Also why do we need to keep the extra sumCoin variable? It looks like we do the sum and assign it to coinMeta.amount

Reply:
- `coinBalanceChange` | `transferObject `| `moveEvent` are all different event
-  coinMeta.amount may be empty

Reply:
But still being different events doesn't answer the question. We have one coinMeta object and we update it using the array of the different events. (A transaction can have multiple events and different combinations of events right?)

> coinMeta.amount may be empty
Oh I see